### PR TITLE
Clarified branching and versioning for plugins.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

I attempted to clarify what we're doing right now for 1.1.
  
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
@saratvemulapalli care to take a look please?

File: RELEASING.md [hunk 1]
```diff
@@ -12,16 +14,32 @@ This document explains the release strategy for artifacts in this organization.
...
+
+* **main**: The _next major_ release, currently 2.0. This is the branch where all merges take place and code moves fast.
+* **1.x**: The _next minor_ release, currently 1.1. Once a change is merged into `main`, decide whether to backport it to `1.x`.
+* **1.0**: The _current_ release, currently 1.0. In between minor releases, only hotfixes (e.g. security) are backported to `1.0`. The next release out of this branch will be 1.0.1.
 
 Label PRs with the next major version label (e.g. `2.0.0`) and merge changes into `main`. Label PRs that you believe need to be backported as `1.x` and `1.0`. Backport PRs by checking out the versioned branch, cherry-pick changes and open a PR against each target backport branch.
 
+### Plugin Branching
+
+Plugins, such as [job-scheduler](https://github.com/opensearch-project/job-scheduler) aren't as active as OpenSearch, and typically track 2 releases in parallel instead of 3. For example, given the last major release of 1.0, job-scheduler maintains the following active branches.
```
Comment (saratvemulapalli) [inline RELEASING.md:31]: I understand why we'd like to reduce it to 2 branches. 
We should communicate to plugins and remove 1.x branches if they've already created.

**Review by saratvemulapalli — APPROVED** [review]

File: RELEASING.md (hunk 1 above)
Comment (dblock) [reply RELEASING.md:31]: I actually don't think a 1.x branch is a bad idea. It will make things easier to merge. I'll update the text.

Comment (CEHENKLE) [reply RELEASING.md:31]: Can we explicitly add that 1,x is optional for plugins?

**Review by CEHENKLE — APPROVED** [review]

File: RELEASING.md (hunk 1 above)
Comment (dblock) [reply RELEASING.md:31]: But it's not. What's optional is that you _work_ on the 1.x branch to do active development. Which is explained in the example right below this paragraph.

Comment (dblock) [reply RELEASING.md]: You're right. See my update in https://github.com/opensearch-project/.github/pull/33/commits/eb83b9de6b540d5d92564a22450545789212d8a0

**Review by CEHENKLE — APPROVED** [review]

**Pull request merged**